Realm/RLMProperty.mm

```diff
@@ -343,6 +343,9 @@ - (instancetype)initSwiftPropertyWithName:(NSString *)name
     }
     else if ([_objcRawType isEqualToString:@"@\"NSNumber\""]) {
         const char *numberType = [propertyValue objCType];
+        if (!numberType) {
+            @throw RLMException(@"Can't persist NSNumber directly: use a Swift-native number type or provide a default value, which allows to infer the type");
```

Comment:
I agree. This phrasing was suggested to me by https://github.com/realm/realm-cocoa/pull/3767#discussion_r67742399. I'd be okay to drop off "which allows to infer the type" entirely. Thoughts?